### PR TITLE
Set Links to the anchor points in the Help file for the singly used sub-dialogs

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes  second part of #6967

@africanmathsinitiative/developers This is in progress

### CODE REVIEW

**Comment by Wycklife** [comment]
@rdstern I have linked most of the sub-dialogs to their anchor points. This is still in progress but you can as well test these ones as I look forward to linking the rest of the sub-dialogs. I have also uploaded the updated version of the help file. Let me know whether they link well. Thank you.

**Comment by rdstern** [comment]
[chm.zip](https://github.com/africanmathsinitiative/R-Instat/files/7678810/chm.zip)
I looked at the existing file and you seemed to be linking to the sub-help, **_not_** to the anchor points.  And the anchor points were not so clear.
So here is a more up-to-date help.  I have made the anchor points clearer for the One Variable Graph and the One Variable Frequency dialogues.  Their location is close to the bottom of the help page for the dialogue.  
The anchor point for the Describe > One Variable > Graph sub-dialogue is called `GraphOptions`
The anchor point for the Describe > One Variable >Frequencies sub-dialogue is called `One`
The anchor point for the Describe > Multivariate > Correlations sub-dialogue is called `Correlation`
The anchor point for the File > Import and Tidy netcdf  sub-dialogue is called `NetCDF` 
The anchor point for the Climatic > Wind Rose dialogue is called `Wind`

**Comment by Wycklife** [comment]
@rdstern I have tried to access the file, and it says "Not Found". Could you share the file via skype?

**Review by rdstern — CHANGES_REQUESTED** [review]
@Wycklife I checked the One Variable Graph and also one Variable Frequencies.
It now goes to the correct help, but at the top, not at the anchor point for the sub-dialogue.  I assume you are using the instructions I forwarded from the helpndoc correspondence (to you and Maxwell) when you started the work.

**Comment by maxwellfundi** [comment]
@rdstern  we are looking at this. Thanks

**Comment by maxwellfundi** [comment]
@dannyparsons  on this issue of the help to show anchor points, we currently don't have a field that can show the anchor points in the form of  `"ms-its:YOUR_SECOND_HELP_FILE.chm::/HELP_ID.htm#ANCHOR_NAME"`
Where "YOUR_SECOND_HELP_FILE" is the name of the help file, "HELP_ID" is the topic to link to and "ANCHOR_NAME" with the name of the anchor to link to.
Quick thoughts is create a new field in the help class that's has these parameters. Any thoughts? Available for a quick call today on this?

**Comment by dannyparsons** [comment]
I'm not sure what you mean by the "help class". I don't think we have our own Help class, we are setting the HelpID in `ucrButtons` and then calling the `Help.ShowHelp` function.

If you mean adding another field to `ucrButtons` for the Anchor name, I think that's a good idea.

You then just need to figure out where to place the Anchor name in the call to `Help.ShowHelp()`

**Comment by Wycklife** [comment]
@rdstern I have tried finding a solution to this PR but I'm still unable to fix it. We discussed with @N-thony  where we even wrote to the HelpNDoc people asking them whether it's possible to set the link in vb.Net, up to now they are still unsure on how that should be done. If @dannyparsons has got any idea on how to do this then I will be happy to discuss it with him still. Meanwhile @N-thony may be willing to share the feedback we received from them.

**Comment by N-thony** [comment]
> @rdstern I have tried finding a solution to this PR but I'm still unable to fix it. We discussed with @N-thony where we even wrote to the HelpNDoc people asking them whether it's possible to set the link in vb.Net, up to now they are still unsure on how that should be done. If @dannyparsons has got any idea on how to do this then I will be happy to discuss it with him still. Meanwhile @N-thony may be willing to share the feedback we received from them.

@Wycklife yes, this is the response from the HelpNDoc, we should find another solution for this problem.
image

**Comment by rdstern** [comment]
Ok, we won 't link from R-Instat to anchor points then!

**Comment by Wycklife** [comment]
@Patowhiz and @Wycklife looked into this further.

@rdstern Yes we can link to the anchor points. We just needed to interpret the email screenshot above correctly. We are using the .Net [Help.ShowHelp](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.help.showhelp?view=windowsdesktop-6.0#system-windows-forms-help-showhelp(system-windows-forms-control-system-string)) subroutine for our help functionality, which doesn't directly support passing the url format shown in the email. 
To support the anchor points we need to modify our current code as shown below;

**Current code**
 `Help.ShowHelp(Me.Parent, frmMain.strStaticPath & "\" & frmMain.strHelpFilePath, HelpNavigator.TopicId, iHelpTopicID.ToString())`

**Modified Code**
 `Help.ShowHelp(Me.Parent, k, HelpNavigator.Topic, "Frequencies.htm#One")`

As you can see above, we just needed to supply the anchor point(`#One`) as part of the topic identifier.
Currently, we are linking the help pages using the `Help context id` instead of the `Help id` (see screenshot below). The `Help.ShowHelp()` subroutine doesn't support linking to anchor points using `Help context id` as the topic identifier. We think this is because the 'internal' pages are saved with `Help id` as their names.
help_page

So basically, to achieve the anchor functionality, we think there are 2 possible solutions;
1. Use the same number used in the `Help context id` as the `Help id` - we recommend this.
2. Use the `Help id` as the topic identifier in VB.Net. - this requires some slight modifications for the dialogs that need this feature.

@rdstern @dannyparsons what are thoughts?

**Comment by rdstern** [comment]
@Wycklife I am now confused about these tow pull requests.  I now assume it is this one that can simply be closed?  Am I right?  If so, what is the situation with #7145 ?  Is that doing something different, or should they both be closed?

**Comment by Wycklife** [comment]
After discussing with @Patowhiz and @rdstern we concluded that this PR should be closed since we are currently unable to link to anchor points.

**Comment by rdstern** [comment]
Ok, I am closing it.

**Comment by Wycklife** [comment]
> @Wycklife I am now confused about these tow pull requests. I now assume it is this one that can simply be closed? Am I right? If so, what is the situation with #7145 ? Is that doing something different, or should they both be closed?

This one should be closed.
